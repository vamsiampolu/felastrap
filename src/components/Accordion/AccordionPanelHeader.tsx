import { useContext } from "react";
import { useFela } from "react-fela";
import { Header, type HeaderProps } from "../Header/Header";
import { AccordionPanelContext } from "./AccordionPanelContext";
import { ToggleButton } from "./ToggleButton";
import { ToggleIcon } from "./ToggleIcon";
import { OnClick } from "../common";

export const AccordionPanelHeader = ({
  className = "",
  children,
}: Omit<HeaderProps, "as">) => {
  const { css } = useFela();
  const { first, last, toggle, isOpen } = useContext(AccordionPanelContext)!;
  const cx = `${css({
    marginBottom: 0,
    marginTop: 0,
  })} accordion-header ${className}`;

  return (
    <Header
      data-testid="accordion-panel-header"
      as="h2"
      className={cx}
      onClick={toggle as OnClick<"h2">}
    >
      <ToggleButton first={first} last={last} isOpen={isOpen}>
        {children}
        <ToggleIcon isOpen={isOpen} />
      </ToggleButton>
    </Header>
  );
};
