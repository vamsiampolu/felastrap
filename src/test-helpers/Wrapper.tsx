import type { IRenderer } from "fela";
import { type PropsWithChildren } from "react";
import { RendererProvider, ThemeProvider } from "react-fela";

const Wrapper = ({
  renderer,
  theme = {},
  children,
}: PropsWithChildren<{ renderer: IRenderer; theme?: object }>) => (
  <ThemeProvider theme={theme}>
    <RendererProvider renderer={renderer}>{children}</RendererProvider>;
  </ThemeProvider>
);

export { Wrapper };
