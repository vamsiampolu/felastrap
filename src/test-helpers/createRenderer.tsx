import expandShorthand from "fela-plugin-expand-shorthand";
import sortClassNames from "fela-sort-classnames";
import sortMediaQueryMobileFirst from "fela-sort-media-query-mobile-first";
import prefixer from "fela-plugin-prefixer";
import placeholderPrefixer from "fela-plugin-placeholder-prefixer";
import { createRenderer as createFelaRenderer, type IRenderer } from "fela";
import { lightTheme } from "../reboot/lightTheme";

export const createRenderer = () => {
  const autoMergeLonghands = true;
  const enhancers = [sortClassNames(), sortMediaQueryMobileFirst()];
  const plugins = [
    prefixer(),
    placeholderPrefixer(),
    expandShorthand(autoMergeLonghands),
  ];

  return createFelaRenderer({
    plugins,
    enhancers,
  });
};

export const cssReboot = (renderer: IRenderer) => {
  renderer.renderStatic(lightTheme);
};
