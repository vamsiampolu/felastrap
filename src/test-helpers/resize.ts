export const resizeWidth = (width: number) => {
  window.innerWidth = width;
  dispatchEvent(new Event("resize"));
};

export const resizeHeight = (height: number) => {
  window.innerHeight = height;
  dispatchEvent(new Event("resize"));
};
