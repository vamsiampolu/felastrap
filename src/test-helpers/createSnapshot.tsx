import { IRenderer } from "fela";
import { renderToString } from "fela-tools";
import { ReactElement, ReactNode } from "react";

import { render, screen } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import {
  type UserEvent,
  type UserEventApi,
} from "@testing-library/user-event/dist/types/setup/setup";
import * as prettier from "prettier";
import { renderToStaticMarkup } from "react-dom/server";
import { Wrapper } from "./Wrapper";

async function formatCSS(css: string) {
  return prettier.format(css, {
    parser: "css",
    useTabs: false,
    tabWidth: 2,
  });
}

function formatHTML(str: string) {
  return prettier.format(str, { parser: "html" });
}

function renderToMarkup(component: ReactElement): string {
  return renderToStaticMarkup(component);
}

export const createSnapshot = async ({
  component,
  renderer,
  theme = {},
}: {
  component: ReactNode;
  renderer: IRenderer;
  theme?: object;
}) => {
  // reset renderer to have a clean setup
  renderer.clear();

  const element = (
    <Wrapper theme={theme} renderer={renderer}>
      {component}
    </Wrapper>
  );

  const view = renderToMarkup(element);
  const str = renderToString(renderer);

  const css = await formatCSS(str);
  const html = await formatHTML(view);

  return `${css}\n\n${html}`;
};

export type EventName = keyof UserEventApi | "focus" | "focus-visible";
export type CreateUserEventType = {
  eventName: EventName;
  component: ReactElement;
  renderer: IRenderer;
  testId: string;
  theme?: object;
  donotWrap?: boolean;
};

export type SetupEvent = {
  user: UserEvent;
  eventName: EventName;
  element: HTMLElement;
};
const setupEvent = async ({ user, eventName, element }: SetupEvent) => {
  switch (eventName) {
    case "hover": {
      await user.hover(element);
      break;
    }

    case "focus": {
      element.focus();
      break;
    }

    case "focus-visible": {
      element.contentEditable = "true";
      element.focus();
      element.contentEditable = "false";
      break;
    }
  }
};

const clearEvent = async ({ eventName, user, element }: SetupEvent) => {
  switch (eventName) {
    case "hover": {
      await user.unhover(element);
      break;
    }

    case "focus":
    case "focus-visible": {
      element.blur();
      break;
    }
  }
};

export const createUserEventSnapshot = async ({
  eventName,
  component,
  testId,
  renderer,
  theme = {},
  donotWrap = false,
}: CreateUserEventType) => {
  renderer.clear();

  const user = userEvent.setup();
  let wrappedComponent = component;
  if (!donotWrap) {
    wrappedComponent = (
      <Wrapper theme={theme} renderer={renderer}>
        {component}
      </Wrapper>
    );
  }

  render(wrappedComponent);

  const element = await screen.findByTestId(testId);

  setupEvent({ element, eventName, user });
  const view = await renderToMarkup(wrappedComponent);
  const utils = renderToString(renderer);
  const css = await formatCSS(utils);
  const html = await formatHTML(view);

  clearEvent({ eventName, element, user });

  return `${css}\n\n${html}`;
};
