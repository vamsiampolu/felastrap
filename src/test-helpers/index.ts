import { Wrapper } from "./Wrapper";
import { createRenderer, cssReboot } from "./createRenderer";
import { createSnapshot, createUserEventSnapshot } from "./createSnapshot";
import { resizeHeight, resizeWidth } from "./resize";

export {
  createRenderer,
  cssReboot,
  createSnapshot,
  createUserEventSnapshot,
  resizeWidth,
  resizeHeight,
  Wrapper,
};
