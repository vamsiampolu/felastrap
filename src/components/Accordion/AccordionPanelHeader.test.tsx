import { IRenderer } from "fela";
import {
  AccordionPanelContext,
  type AccordionPanelContextValue,
} from "./AccordionPanelContext";
import { AccordionPanelHeader } from "./AccordionPanelHeader";
import { Wrapper, createRenderer, createSnapshot } from "../../test-helpers";
import { render, screen } from "@testing-library/react";

describe("components/Accordion/AccordionPanelHeader", () => {
  let renderer: IRenderer;

  beforeAll(() => {
    renderer = createRenderer();
  });

  beforeEach(() => {
    renderer.clear();
  });

  afterAll(() => {
    renderer = undefined as unknown as IRenderer;
  });

  const value: AccordionPanelContextValue = {
    first: false,
    last: false,
    toggle: () => undefined,
    isOpen: false,
  };

  it("should render a panel header", async () => {
    const component = (
      <AccordionPanelContext.Provider value={value}>
        <AccordionPanelHeader>Panel Header</AccordionPanelHeader>
      </AccordionPanelContext.Provider>
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("accordion-panel-header")).toBeInTheDocument();
    expect(screen.getByTestId("accordion-panel-header")).toHaveClass(
      "accordion-header",
    );

    expect(screen.getByTestId("accordion-panel-header")?.tagName).toEqual("H2");
    expect(screen.getByTestId("accordion-panel-header")).toHaveTextContent(
      "Panel Header",
    );

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should render a toggle button and the toggle icon", async () => {
    const component = (
      <AccordionPanelContext.Provider value={value}>
        <AccordionPanelHeader>Panel Header</AccordionPanelHeader>
      </AccordionPanelContext.Provider>
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("accordion-panel-header")).toBeInTheDocument();
    expect(screen.getByTestId("accordion-toggle-icon")).toBeInTheDocument();

    expect(screen.getByTestId("accordion-toggle-button")).toBeInTheDocument();
    expect(screen.getByTestId("accordion-toggle-button")).toHaveTextContent(
      "Panel Header",
    );

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });
});
