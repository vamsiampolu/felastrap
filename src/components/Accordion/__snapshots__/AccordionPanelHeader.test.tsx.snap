// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Accordion/AccordionPanelHeader should render a panel header 1`] = `
".a {
  margin-bottom: 0;
}
.b {
  margin-top: 0;
}
.c {
  color: inherit;
}
.d {
  font-weight: 500;
}
.e {
  font-size: calc(1.325rem + 0.9vw);
}
.f {
  line-height: 1.2;
}
.g {
  margin-bottom: 0.5rem;
}
.i {
  position: relative;
}
.j {
  display: flex !important;
}
.k {
  align-items: center;
}
.l {
  width: 100%;
}
.m {
  font-size: 1rem;
}
.n {
  color: #212529;
}
.o {
  text-align: left;
}
.p {
  background-color: #fff;
}
.q {
  overflow-anchor: none;
}
.af {
  transition-property: color, background-color, border-color, box-shadow,
    border-radius;
}
.ag {
  transition-duration: 0.15s, 0.15s, 0.15s, 0.15s, 0.15s;
}
.ah {
  transition-delay: ease-in-out, ease-in-out, ease-in-out, ease-in-out, ease;
}
.ai {
  transition-timing-function: ease, ease, ease, ease, ease;
}
.aj {
  border-top-left-radius: 0;
}
.ak {
  border-top-right-radius: 0;
}
.al {
  border-bottom-right-radius: 0;
}
.am {
  border-bottom-left-radius: 0;
}
.an {
  border-top-width: 0;
}
.ao {
  border-right-width: 0;
}
.ap {
  border-bottom-width: 0;
}
.aq {
  border-left-width: 0;
}
.ar {
  padding-top: 1.25rem;
}
.as {
  padding-right: 1rem;
}
.at {
  padding-bottom: 1.25rem;
}
.au {
  padding-left: 1rem;
}
.av {
  border-top-left-radius: 4px;
}
.aw {
  border-bottom-left-radius: 4px;
}
.ax {
  padding-top: 6px;
}
.ay {
  padding-bottom: 6px;
}
.az {
  padding-left: 12px;
}
.ba {
  padding-right: 12px;
}
.bb {
  background-image: none;
}
.bc {
  color: #333;
}
.bd {
  cursor: pointer;
}
.be {
  display: inline-block;
}
.bf {
  font-size: 14px;
}
.bg {
  font-weight: 400;
}
.bh {
  line-height: 1.42857143;
}
.bi {
  text-align: center;
}
.bj {
  touch-action: manipulation;
}
.bk {
  white-space: nowrap;
}
.bl {
  vertical-align: middle;
}
.bm {
  user-select: none;
}
.bw:disabled {
  background-color: #fff;
}
.bx:disabled {
  color: #333;
}
.by:disabled {
  box-shadow: none;
}
.bz:disabled {
  cursor: not-allowed;
}
.ca:disabled {
  filter: alpha(opacity=65);
}
.cb:disabled {
  opacity: 0.65;
}
.cc:disabled {
  pointer-events: none;
}
.cd:disabled {
  border-top-color: #ccc;
}
.ce:disabled {
  border-right-color: #ccc;
}
.cf:disabled {
  border-bottom-color: #ccc;
}
.cg:disabled {
  border-left-color: #ccc;
}
.cv {
  border-top-width: 1px;
}
.cw {
  border-right-width: 1px;
}
.cx {
  border-bottom-width: 1px;
}
.cy {
  border-left-width: 1px;
}
.cz {
  border-top-style: solid;
}
.da {
  border-right-style: solid;
}
.db {
  border-bottom-style: solid;
}
.dc {
  border-left-style: solid;
}
.dd {
  border-top-color: #ccc;
}
.de {
  border-right-color: #ccc;
}
.df {
  border-bottom-color: #ccc;
}
.dg {
  border-left-color: #ccc;
}
.dh {
  flex-shrink: 0;
}
.di {
  width: 1.25rem;
}
.dj {
  height: 1.25rem;
}
.dk {
  margin-left: auto;
}
.dl {
  background-image: url("/chevron-down.svg");
}
.dm {
  background-repeat: no-repeat;
}
.dn {
  background-size: 1.25rem;
}
.do {
  transition-property: transform;
}
.dp {
  transition-duration: 0.2s;
}
.dq {
  transition-delay: ease-in-out;
}
.dr {
  transition-timing-function: ease;
}
.ae:hover {
  z-index: 2;
}
.cp:hover {
  background-color: #e6e6e6;
}
.cq:hover {
  color: #333;
}
.cr:hover {
  border-top-color: #adadad;
}
.cs:hover {
  border-right-color: #adadad;
}
.ct:hover {
  border-bottom-color: #adadad;
}
.cu:hover {
  border-left-color: #adadad;
}
.v:focus {
  z-index: 3;
}
.w:focus {
  box-shadow: 0 0 0 0.25rem rgba(13, 110, 253, 0.25);
}
.x:focus {
  outline-width: 0;
}
.y:focus {
  border-top-color: #86b7fe;
}
.z:focus {
  border-right-color: #86b7fe;
}
.ab:focus {
  border-bottom-color: #86b7fe;
}
.ac:focus {
  border-left-color: #86b7fe;
}
.ch:focus-visible {
  color: #333;
}
.ci:focus-visible {
  background-color: #e6e6e6;
}
.cj:focus-visible {
  box-shadow: 0 0 0 0.25rem rgba(0, 0, 0, 0.125);
}
.ck:focus-visible {
  outline-width: 0;
}
.cl:focus-visible {
  border-top-color: #adadad;
}
.cm:focus-visible {
  border-right-color: #adadad;
}
.cn:focus-visible {
  border-bottom-color: #adadad;
}
.co:focus-visible {
  border-left-color: #adadad;
}
.bn:active {
  color: #333;
}
.bo:active {
  background-color: #e6e6e6;
}
.bp:active {
  background-image: none;
}
.bq:active {
  box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
}
.br:active {
  outline-width: 0;
}
.bs:active {
  border-top-color: #adadad;
}
.bt:active {
  border-right-color: #adadad;
}
.bu:active {
  border-bottom-color: #adadad;
}
.bv:active {
  border-left-color: #adadad;
}
@media (min-width: 1200px) {
  .h {
    font-size: 2rem;
  }
}
@media (prefers-reduced-motion: reduce) {
  .r {
    transition-property: none;
  }
  .s {
    transition-duration: 0s;
  }
  .t {
    transition-delay: 0s;
  }
  .u {
    transition-timing-function: ease;
  }
}


<h2 class="a b accordion-header" data-testid="accordion-panel-header">
  <button
    data-testid="accordion-toggle-button"
    class="a ak al av aw ax ay az ba bb bc bd be bf bg bh bi bj bk bl bm bn bo bp bq br bs bt bu bv bw bx by bz ca cb cc cd ce cf cg ch ci cj ck cl cm cn co cp cq cr cs ct cu cv cw cx cy cz da db dc dd de df dg p ab ac ae af ag ah ai aj ak al am an ao ap aq ar as at au i j k l m n o p q r s t u v w x y z accordion-toggle-button"
    aria-expanded="false"
  >
    Panel Header
    <div
      data-testid="accordion-toggle-icon"
      class="dh di dj dk dl dm dn do dp dq dr r s t u accordion-toggle-icon"
    ></div>
  </button>
</h2>
;
"
`;

exports[`components/Accordion/AccordionPanelHeader should render a toggle button and the toggle icon 1`] = `
".a {
  margin-bottom: 0;
}
.b {
  margin-top: 0;
}
.c {
  color: inherit;
}
.d {
  font-weight: 500;
}
.e {
  font-size: calc(1.325rem + 0.9vw);
}
.f {
  line-height: 1.2;
}
.g {
  margin-bottom: 0.5rem;
}
.i {
  position: relative;
}
.j {
  display: flex !important;
}
.k {
  align-items: center;
}
.l {
  width: 100%;
}
.m {
  font-size: 1rem;
}
.n {
  color: #212529;
}
.o {
  text-align: left;
}
.p {
  background-color: #fff;
}
.q {
  overflow-anchor: none;
}
.af {
  transition-property: color, background-color, border-color, box-shadow,
    border-radius;
}
.ag {
  transition-duration: 0.15s, 0.15s, 0.15s, 0.15s, 0.15s;
}
.ah {
  transition-delay: ease-in-out, ease-in-out, ease-in-out, ease-in-out, ease;
}
.ai {
  transition-timing-function: ease, ease, ease, ease, ease;
}
.aj {
  border-top-left-radius: 0;
}
.ak {
  border-top-right-radius: 0;
}
.al {
  border-bottom-right-radius: 0;
}
.am {
  border-bottom-left-radius: 0;
}
.an {
  border-top-width: 0;
}
.ao {
  border-right-width: 0;
}
.ap {
  border-bottom-width: 0;
}
.aq {
  border-left-width: 0;
}
.ar {
  padding-top: 1.25rem;
}
.as {
  padding-right: 1rem;
}
.at {
  padding-bottom: 1.25rem;
}
.au {
  padding-left: 1rem;
}
.av {
  border-top-left-radius: 4px;
}
.aw {
  border-bottom-left-radius: 4px;
}
.ax {
  padding-top: 6px;
}
.ay {
  padding-bottom: 6px;
}
.az {
  padding-left: 12px;
}
.ba {
  padding-right: 12px;
}
.bb {
  background-image: none;
}
.bc {
  color: #333;
}
.bd {
  cursor: pointer;
}
.be {
  display: inline-block;
}
.bf {
  font-size: 14px;
}
.bg {
  font-weight: 400;
}
.bh {
  line-height: 1.42857143;
}
.bi {
  text-align: center;
}
.bj {
  touch-action: manipulation;
}
.bk {
  white-space: nowrap;
}
.bl {
  vertical-align: middle;
}
.bm {
  user-select: none;
}
.bw:disabled {
  background-color: #fff;
}
.bx:disabled {
  color: #333;
}
.by:disabled {
  box-shadow: none;
}
.bz:disabled {
  cursor: not-allowed;
}
.ca:disabled {
  filter: alpha(opacity=65);
}
.cb:disabled {
  opacity: 0.65;
}
.cc:disabled {
  pointer-events: none;
}
.cd:disabled {
  border-top-color: #ccc;
}
.ce:disabled {
  border-right-color: #ccc;
}
.cf:disabled {
  border-bottom-color: #ccc;
}
.cg:disabled {
  border-left-color: #ccc;
}
.cv {
  border-top-width: 1px;
}
.cw {
  border-right-width: 1px;
}
.cx {
  border-bottom-width: 1px;
}
.cy {
  border-left-width: 1px;
}
.cz {
  border-top-style: solid;
}
.da {
  border-right-style: solid;
}
.db {
  border-bottom-style: solid;
}
.dc {
  border-left-style: solid;
}
.dd {
  border-top-color: #ccc;
}
.de {
  border-right-color: #ccc;
}
.df {
  border-bottom-color: #ccc;
}
.dg {
  border-left-color: #ccc;
}
.dh {
  flex-shrink: 0;
}
.di {
  width: 1.25rem;
}
.dj {
  height: 1.25rem;
}
.dk {
  margin-left: auto;
}
.dl {
  background-image: url("/chevron-down.svg");
}
.dm {
  background-repeat: no-repeat;
}
.dn {
  background-size: 1.25rem;
}
.do {
  transition-property: transform;
}
.dp {
  transition-duration: 0.2s;
}
.dq {
  transition-delay: ease-in-out;
}
.dr {
  transition-timing-function: ease;
}
.ae:hover {
  z-index: 2;
}
.cp:hover {
  background-color: #e6e6e6;
}
.cq:hover {
  color: #333;
}
.cr:hover {
  border-top-color: #adadad;
}
.cs:hover {
  border-right-color: #adadad;
}
.ct:hover {
  border-bottom-color: #adadad;
}
.cu:hover {
  border-left-color: #adadad;
}
.v:focus {
  z-index: 3;
}
.w:focus {
  box-shadow: 0 0 0 0.25rem rgba(13, 110, 253, 0.25);
}
.x:focus {
  outline-width: 0;
}
.y:focus {
  border-top-color: #86b7fe;
}
.z:focus {
  border-right-color: #86b7fe;
}
.ab:focus {
  border-bottom-color: #86b7fe;
}
.ac:focus {
  border-left-color: #86b7fe;
}
.ch:focus-visible {
  color: #333;
}
.ci:focus-visible {
  background-color: #e6e6e6;
}
.cj:focus-visible {
  box-shadow: 0 0 0 0.25rem rgba(0, 0, 0, 0.125);
}
.ck:focus-visible {
  outline-width: 0;
}
.cl:focus-visible {
  border-top-color: #adadad;
}
.cm:focus-visible {
  border-right-color: #adadad;
}
.cn:focus-visible {
  border-bottom-color: #adadad;
}
.co:focus-visible {
  border-left-color: #adadad;
}
.bn:active {
  color: #333;
}
.bo:active {
  background-color: #e6e6e6;
}
.bp:active {
  background-image: none;
}
.bq:active {
  box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
}
.br:active {
  outline-width: 0;
}
.bs:active {
  border-top-color: #adadad;
}
.bt:active {
  border-right-color: #adadad;
}
.bu:active {
  border-bottom-color: #adadad;
}
.bv:active {
  border-left-color: #adadad;
}
@media (min-width: 1200px) {
  .h {
    font-size: 2rem;
  }
}
@media (prefers-reduced-motion: reduce) {
  .r {
    transition-property: none;
  }
  .s {
    transition-duration: 0s;
  }
  .t {
    transition-delay: 0s;
  }
  .u {
    transition-timing-function: ease;
  }
}


<h2 class="a b accordion-header" data-testid="accordion-panel-header">
  <button
    data-testid="accordion-toggle-button"
    class="a ak al av aw ax ay az ba bb bc bd be bf bg bh bi bj bk bl bm bn bo bp bq br bs bt bu bv bw bx by bz ca cb cc cd ce cf cg ch ci cj ck cl cm cn co cp cq cr cs ct cu cv cw cx cy cz da db dc dd de df dg p ab ac ae af ag ah ai aj ak al am an ao ap aq ar as at au i j k l m n o p q r s t u v w x y z accordion-toggle-button"
    aria-expanded="false"
  >
    Panel Header
    <div
      data-testid="accordion-toggle-icon"
      class="dh di dj dk dl dm dn do dp dq dr r s t u accordion-toggle-icon"
    ></div>
  </button>
</h2>
;
"
`;
